repo.c

```diff
@@ -531,6 +543,113 @@ download_repo_metadata (SoupSession *session,
   return fpath;
 }
 
+static void
+switch_to_cached_repo (Repo        *repo,
+                       Repodata    *repodata,
+                       const char  *repoext,
+                       const char  *cachepath)
+{
+  int i;
+  FILE *fp;
+
+  if (!repoext && repodata)
+    return;
+
+  for (i = repo->start; i < repo->end; i++)
+    if (repo->pool->solvables[i].repo != repo)
+      break;
+
+  if (i < repo->end)
+    return; /* not a simple block */
+
+  fp = g_fopen (cachepath, "r");
+  if (!fp)
+    return;
+
+  /* main repo */
+  if (!repoext)
+    {
+      repo_empty (repo, 1);
+      if (repo_add_solv (repo, fp, SOLV_ADD_NO_STUBS))
+        {
+          g_warning ("Could not add solvables from cache file");
+          return;
+        }
+    }
+  else
+    {
+      int flags = REPO_USE_LOADING | REPO_EXTEND_SOLVABLES | REPO_LOCALPOOL;
+      /* make sure repodata contains complete repo */
+      /* (this is how repodata_write saves it) */
+      repodata_extend_block (repodata, repo->start, repo->end - repo->start);
+      repodata->state = REPODATA_LOADING;
+      repo_add_solv (repo, fp, flags);
+      repodata->state = REPODATA_AVAILABLE; /* in case the load failed */
+    }
+
+  fclose (fp);
+}
+
+static gboolean
+write_repo_cache (Repo         *repo,
+                  Repodata     *repodata,
+                  const char   *repoext,
+                  const gchar  *cachename)
+{
+  FILE *fp = g_fopen (cachename, "w");
+  if (!fp)
+    {
+      g_warning ("Could not open cache file %s: %s", cachename, g_strerror (errno));
+      return FALSE;
+    }
+
+  if (fchmod (fileno (fp), 0444) < 0)
```

Comment:
why do you change permissions?

Reply:
No special reason. Just thought it'd be good to make the file read-only since it's managed by fus. I'll remove it.